logger_core/src/lib.rs

```diff
@@ -117,6 +118,25 @@ impl Level {
     }
 }
 
+/// Attempt to read a directory path from an environment variable. If the environment variable `envname` exists
+/// and contains a valid path - this function will create and return that path. In any case of failure,
+/// this method returns `None` (e.g. the environment variable exists but contains an empty path etc)
+pub fn directory_from_env(envname: &str) -> Option<String> {
```

Comment:
I think the function name dont fits the logic and it might contains two logics in one.
The function is generic, you dont take dir from env, you take what i give to the function from env; then perform a specific to dir logic which is trim, and create.

If you want it to be specific put everything inside, including the specific `envname` and what to return otherwise.
If its should be generic, the name dont fits, and its should be used by the get dir function, which also check the env using it.

Reply:
If i pass `RUST_LOG` to the function i can get log dir `trace`.

Reply:
Looking at the rest of the code it should be specific, making it accepting variable is error prone, but not save logic.

Reply:
Yes, I actually meant to rename it to: `create_dir_from_env` to make it more explicit as it hides its actual action (creating the directory)